accepted/2021/statics-in-interfaces/README.md

```diff
@@ -8,6 +8,791 @@ C# is looking at enabling static abstract members in interfaces (https://github.
+
+        // TODO: The Exponent and Signifcand need to be exposed here
+        // They should return INumber<TSelf> here and IBinaryNumber<TSelf> on IBinaryFloatingPoint<TSelf>
+
+        // TODO: We may want to expose constants related to the subnormal and finite boundaries
+
+        // The following constants are defined by System.Math and System.MathF today
+
+        static abstract TSelf E { get; }
+
+        static abstract TSelf PI { get; }
+
+        static abstract TSelf Tau { get; }
+
+        // The following methods are defined by System.Math and System.MathF today
+        // Exposing them on the interfaces means they will become available as statics on the primitive types
+        // This, in a way, obsoletes Math/MathF and brings float/double inline with how non-primitive types support similar functionality
+        // API review will need to determine if we'll want to continue adding APIs to Math/MathF in the future
+
+        static TSelf Acos(TSelf x);
+
+        static TSelf Acosh(TSelf x);
+
+        static TSelf Asin(TSelf x);
+
+        static TSelf Asinh(TSelf x);
+
+        static TSelf Atan(TSelf x);
+
+        static TSelf Atan2(TSelf y, TSelf x);
+
+        static TSelf Atanh(TSelf x);
+
+        static TSelf BitIncrement(TSelf x);
+
+        static TSelf BitDecrement(TSelf x);
+
+        static TSelf Cbrt(TSelf x);
+
+        static TSelf Ceiling(TSelf x);
+
+        // CopySign is likely more general purpose and may apply to any signed number type
+        // It may even be applicable to unsigned numbers where it simply returns x, for convenience
+
+        static TSelf CopySign(TSelf x, TSelf y);
+
+        static TSelf Cos(TSelf x);
+
+        static TSelf Cosh(TSelf x);
+
+        static TSelf Exp(TSelf x);
+
+        static TSelf Floor(TSelf x);
+
+        static TSelf FusedMultiplyAdd(TSelf x, TSelf y, TSelf z);
+
+        static TSelf IEEERemainder(TSelf x, TSelf y);
+
+        // IEEE defines the result to be an integral type, but not the size
+
+        static int ILogB(TSelf x);
+
+        static TSelf Log(TSelf x);
+
+        static TSelf Log(TSelf x, TSelf newBase);
+
+        static TSelf Log2(TSelf x);
+
+        static TSelf Log10(TSelf x);
+
+        static TSelf MaxMagnitude(TSelf x, TSelf y);
+
+        static TSelf MinMagnitude(TSelf x, TSelf y);
+
+        static TSelf Pow(TSelf x, TSelf y);
+
+        static TSelf Round(TSelf x);
+
+        static TSelf Round(TSelf x, int digits);
+
+        static TSelf Round(TSelf x, MidpointRounding mode);
+
+        static TSelf Round(TSelf x, int digits, MidpointRounding mode);
+
+        // IEEE defines n to be an integral type, but not the size
+
+        static TSelf ScaleB(TSelf x, int n);
+
+        static TSelf Sin(TSelf x);
+
+        static TSelf Sinh(TSelf x);
+
+        static TSelf Sqrt(TSelf x);
+
+        static TSelf Tan(TSelf x);
+
+        static TSelf Tanh(TSelf x);
+
+        static TSelf Truncate(TSelf x);
```

Comment:
`abstract` is missing from these members.

```suggestion
        static abstract TSelf Acos(TSelf x);

        static abstract TSelf Acosh(TSelf x);

        static abstract TSelf Asin(TSelf x);

        static abstract TSelf Asinh(TSelf x);

        static abstract TSelf Atan(TSelf x);

        static abstract TSelf Atan2(TSelf y, TSelf x);

        static abstract TSelf Atanh(TSelf x);

        static abstract TSelf BitIncrement(TSelf x);

        static abstract TSelf BitDecrement(TSelf x);

        static abstract TSelf Cbrt(TSelf x);

        static abstract TSelf Ceiling(TSelf x);

        // CopySign is likely more general purpose and may apply to any signed number type
        // It may even be applicable to unsigned numbers where it simply returns x, for convenience

        static abstract TSelf CopySign(TSelf x, TSelf y);

        static abstract TSelf Cos(TSelf x);

        static abstract TSelf Cosh(TSelf x);

        static abstract TSelf Exp(TSelf x);

        static abstract TSelf Floor(TSelf x);

        static abstract TSelf FusedMultiplyAdd(TSelf x, TSelf y, TSelf z);

        static abstract TSelf IEEERemainder(TSelf x, TSelf y);

        // IEEE defines the result to be an integral type, but not the size

        static abstract int ILogB(TSelf x);

        static abstract TSelf Log(TSelf x);

        static abstract TSelf Log(TSelf x, TSelf newBase);

        static abstract TSelf Log2(TSelf x);

        static abstract TSelf Log10(TSelf x);

        static abstract TSelf MaxMagnitude(TSelf x, TSelf y);

        static abstract TSelf MinMagnitude(TSelf x, TSelf y);

        static abstract TSelf Pow(TSelf x, TSelf y);

        static abstract TSelf Round(TSelf x);

        static abstract TSelf Round(TSelf x, int digits);

        static abstract TSelf Round(TSelf x, MidpointRounding mode);

        static abstract TSelf Round(TSelf x, int digits, MidpointRounding mode);

        // IEEE defines n to be an integral type, but not the size

        static abstract TSelf ScaleB(TSelf x, int n);

        static abstract TSelf Sin(TSelf x);

        static abstract TSelf Sinh(TSelf x);

        static abstract TSelf Sqrt(TSelf x);

        static abstract TSelf Tan(TSelf x);

        static abstract TSelf Tanh(TSelf x);

        static abstract TSelf Truncate(TSelf x);
```